dask/bag/tests/test_bag.py

```diff
@@ -227,6 +227,24 @@ def test_distinct():
     assert bag.filter(None).distinct().compute() == []
 
 
+def test_distinct_with_key():
+    bag = db.from_sequence([{'a': 0}, {'a': 0}, {'a': 1}])
+    expected = [{'a': 0}, {'a': 1}]
+    key = 'a'
+    key_func = lambda x: x[key]
+    assert bag.distinct(key_func).compute() == expected
+    assert bag.distinct(key).compute() == expected
```

Comment:
I recommend using `assert_eq` rather than `assert` and `compute`.  That runs other checks and also avoids the multiprocessing module, which is slow for windows developers.